stdlib/Markdown/src/render/terminal/render.jl

```diff
@@ -156,20 +173,36 @@ function terminline(io::IO, md::Image)
     terminline(io, "(Image: $(md.alt))")
 end
 
-terminline(io::IO, f::Footnote) = with_output_color(terminline, :bold, io, "[^$(f.id)]")
+function terminline(io::IO, f::Footnote)
+    print(io, styled"{markdown_footnote:[^$(f.id)]}")
+end
 
-function terminline(io::IO, md::Link)
```

Comment:
Shouldn't we leave `terminline(io::IO, md::Link)`?

Reply:
`terminline(::IO, ::Link)` still works, it just goes through `terminline(::IO, ::MarkdownElement)` now.

Reply:
Wouldn't the results be different?

Reply:
Different to what? What was happening before this PR, yea but most things are different in that sense.

Reply:
So, shouldn't there be some note about compatibility?
It is a little difficult to track what has changed in non-`AnnotIO` because there are so many changes.

Reply:
I don't see any compatibility concerns unless people are relying on on the specific rendered output of the `term` / `terminline` functions, which they really shouldn't.

Reply:
I also don't see what this has to do with `AnnotatedIOBuffer`.

Reply:
I do not understand the nuance of whether you are saying there is compatibility or not.

Reply:
```
$ julia --color=no
```
```
julia> using Markdown

julia> md"[x](http://example.com/)"
  x (http://example.com/)
```
```
julia> md"[x](http://example.com/)" # maybe this PR?
  x
```
Yes, since `term` is not public APIs, this is not a breaking change, just a tweak.

Reply:
Right, that is a tweak that's occurred here. If we want to change the output text that will be to be handled via an extra kwarg that's passed though the call chain. Doable, but a bit of a hassle, so I think worth waiting to see if anybody asks for it before doing.

Reply:
Actually, I don't think the above is a `terminline` problem, but more of a fundamental issue.

Reply:
I don't follow.

Reply:
In any case, it is misleading to discuss about the tweak here (i.e., the line `function terminline(io::IO, md::Link)`).

Reply:
Alright, I'll just close this thread then.